vllm/v1/worker/gpu_worker.py

```diff
@@ -209,6 +229,307 @@ def profile(self, is_start=True):
         else:
             self.profiler.stop()
 
+    def check_health(self) -> None:
+        # worker will always be healthy as long as it's running.
+        return
+
+
+@dataclass
+class WorkerInitRequestType:
+    """
+    Request types defined as hex byte strings, so it can be sent over sockets
+    without separate encoding step.
+    """
+    DETERMINE_NUM_BLOCKS = b'\x00'
+    INITIALIZE = b'\x01'  # Initialize cache and begin worker execution
+
+
+@dataclass
+class WorkerInitResponseType:
+    """
+    Request types defined as hex byte strings, so it can be sent over sockets
+    without separate encoding step.
+    """
+    READY = b'\x00'
+    NUM_BLOCKS = b'\x01'
+    INITIALIZE_SUCCESS = b'\x02'
```

Comment:
we should not need this if we set up the connection early, and all these can be normal `collective_rpc` calls.